### PR TITLE
get volunteer mappings

### PR DESCRIPTION
**Pull request by johndeyrup**
This PR will add the last piece so that we can have a manager functionality and functionality so a volunteer can get info about his sites. I will add auth in a later PR.

Examples:
Create a mapping
POST
http://localhost:3002/volunteers-locations
Body:
```
{
    "volunteerId" : 10,
    "locationId": 399
}
```

Get all locations and availibilities a volunteer can check
GET
http://localhost:3002/volunteers-locations/?volunteerId=10&locationId=399
```
[
    {
        "id": 399,
        "name": "Worcester: Worcester Senior Center",
        "bookinglink": "https://www.maimmunizations.org/",
        "address": "128 Providence Street, Worcester, MA 01604",
        "serves": "Eligible populations who live or work in Grafton, Holden, Millbury, Shrewsbury, West Boylston, Worcester",
        "vaccinesoffered": null,
        "siteinstructions": "Parking lot entrance on Spurr Street. Wait in car until appointment. Entrance to EDS clinic side door to main entrance on Providence Street.",
        "daysopen": {},
        "county": "Worcester County",
        "latitude": 42.2483302,
        "longitude": -71.79315369999999,
        "lastupdated": "2021-02-16T02:58:25.030Z",
        "phone": null,
        "accessibility": null,
        "email": null,
        "availability": [
            {
                "id": 7,
                "location_id": 399,
                "doses": 10,
                "availabilitytime": "2020-03-02T00:00:00",
                "lastupdated": "2021-03-03T02:29:51.074005"
            }
        ]
    }
]
```

Delete a mapping for a volunteer
DELETE
http://localhost:3002/volunteers-locations/?volunteerId=10&locationId=399

Fetch volunteer info by email
GET
http://localhost:3002/volunteers/by-email?volunteerEmail=test4@test.com

### CODE REVIEW

File: routes/volunteers.js [hunk 1]
```diff
@@ -61,9 +61,33 @@ router.get('/',
...
+    query('volunteerEmail').isEmail(),
+    async function(req, res) {
+        const errors = validationResult(req);
+        if (!errors.isEmpty()) {
+            return res.status(400).json({ errors: errors.array() });
+        }
+        try {  
+            const volunteers = await volunteersDb.getVolunteerByEmail(req.query.volunteerEmail);
+            res.status(201).json(volunteers);
+        } catch (error) {
```
Comment (mwellman17) [inline routes/volunteers.js:76]: This should be a 200 and return one volunteer, or a 404 if not found.

File: routes/volunteers.js [hunk 2]
```diff
@@ -61,9 +61,33 @@ router.get('/',
...
+            let errorString = `ERROR OCCURRED LOOKING UP LOCATIONS! error: ${error}`;
+            console.log(errorString);
```
Comment (mwellman17) [inline routes/volunteers.js:78]: Nit copypasta. Looking up volunteers, not locations.

**Review by mwellman17 — APPROVED** [review]
NIt, gets and deletes should be 200s and posts should be 201s.

**Review by jonhines — APPROVED** [review]
these new APIs look great.

my only minor comment would be to consider adding the APIs to the postman collection so that we have a documented example of how to use these APIs in one place.

**Comment by johndeyrup** [comment]
:+1: Will get the postman collection in a different PR

**Pull request merged**